lib/ddtrace/profiling/collectors/stack.rb

```diff
@@ -0,0 +1,111 @@
+require 'ddtrace/profiling/events/stack'
+require 'ddtrace/utils/time'
+require 'ddtrace/worker'
+require 'ddtrace/workers/polling'
+
+module Datadog
+  module Profiling
+    module Collectors
+      # Pulls and exports profiling data on an async interval basis
+      class Stack < Worker
+        include Workers::Polling
+
+        MIN_INTERVAL = 0.01
+        MAX_FRAMES = 128
+
+        attr_reader \
+          :ignore_thread,
+          :last_wall_time,
+          :max_frames,
+          :max_time_usage_pct,
+          :recorder
+
+        def initialize(recorder, options = {})
+          @recorder = recorder
+          @max_frames = options.fetch(:max_frames, MAX_FRAMES)
+          @ignore_thread = options.fetch(:ignore_thread, nil)
+          @max_time_usage_pct = options.fetch(:max_time_usage_pct, 2.0)
+
+          # Workers::Async::Thread settings
+          # Restart in forks by default
+          self.fork_policy = options.fetch(:fork_policy, Workers::Async::Thread::FORK_POLICY_RESTART)
```

Comment:
Yes, I'm aware in this case that it doesn't truly auto-restart. The point of the policy is to manage how the worker should behave when called to work when forked.

We simply don't have any `at_fork`/`after_fork` callbacks in the Ruby API by which we can do this natively. (See https://bugs.ruby-lang.org/issues/5446) Whatever does the forking has to call something like `Datadog.after_fork`.

So the plan would to be implement something like that at the `Profiler` level, then have integrations like Resque trigger the callback when their hooks are invoked. It may mean that profiling will not work in forked processes where there isn't a well known callback API.